tests/backend/test_bosonic_backend.py

```diff
@@ -0,0 +1,560 @@
+        backend_plus = bosonic.BosonicBackend()
+        backend_plus.run_prog(prog_plus)
+        state_plus = backend_plus.state()
+        wigner_plus = state_plus.wigner(0,x,p)
+        
+        assert np.allclose(wigner_0H,wigner_plus)
+    
+    @pytest.mark.parametrize("eps", EPS_VALS)
+    def test_gkp_state_parity(self,eps):
+        r"""Checks that GKP states yield the right parity."""    
+        prog = sf.Program(1)
+        with prog.context as q:
+            sf.ops.GKP(epsilon=eps) | q[0]
+              
+        backend = bosonic.BosonicBackend()
+        backend.run_prog(prog)
+        state = backend.state()
+        assert np.allclose(state.parity_expectation([0]),1,atol=r_fock)
+    
+    def test_gkp_complex(self):
+        r"""Checks that trying to call a complex representation
+        raises an error.
+        
+        This test can be updated once the complex representation is
+        implemented."""   
+        with pytest.raises(NotImplementedError):
+            prog = sf.Program(1)
+            with prog.context as q:
+                sf.ops.GKP(desc="complex") | q[0]
+                  
+            backend = bosonic.BosonicBackend()
+            backend.run_prog(prog)
+
+class TestBosonicUserSpecifiedState():
+    """Checks the Bosonic preparation method."""
+    
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    def test_complex_weight(self, alpha):
+        r"""Checks that Bosonic creates a state with user-specifed weights, means
+        and covariances."""
+        dummy_backend = bosonic.BosonicBackend()
+        dummy_backend.begin_circuit(1)
+        # Get weights, means and covs for a cat state
+        weights, means, covs = dummy_backend.prepare_cat(alpha,0,0,'complex',0)
+        
+        # Initiate the state, but use Bosonic method
+        backend = bosonic.BosonicBackend()
+        prog = sf.Program(1)
+        with prog.context as q1:
+            sf.ops.Bosonic(weights,means,covs) | q1[0]
+        backend.run_prog(prog)
+        state = backend.state()
+        
+        #Prepare using the Catstate method
+        backend2 = bosonic.BosonicBackend()
+        prog2 = sf.Program(1)
+        with prog2.context as q2:
+            sf.ops.Catstate(alpha) | q2[0]
+        backend2.run_prog(prog2)
+        state2 = backend2.state()
+        
+        #Check the two states are equal
+        assert state.__eq__(state2)
+        
+class TestBosonicPrograms():
+    """Tests that small programs run and return the correct output."""
+
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    @pytest.mark.parametrize("r", R_VALS)
+    def test_init_circuit(self,alpha,r):
+        """Checks init_circuit only prepares non-Gaussian states and vacuum."""
+        prog = sf.Program(3)
+        with prog.context as q:
+            sf.ops.Catstate(alpha) | q[0]
+            #This line should be ignored since it is a Gaussian prep
+            #that would be picked up in run_prog, but not init_circuit
```

Comment:
```suggestion
            # this line should be ignored since it is a Gaussian prep
            # that would be picked up in run_prog, but not init_circuit
```

Reply:
(minor) but there are a few more occurrences below.